In [ ]:
# Energy and Industrial Sector ETF of S&P 500

In [ ]:
## Introduction

The S&P 500 index is comprised of 500 large1 U.S. public companies traded on an eligible U.S. stock exchange. Its objective is to measure the performance of the large-cap U.S. equity market. These 500 companies are classified into 11 sectors:
* Energy
* Materials
* Industrials
* Consumer discretionary
* Consumer staples
* Health care
* Financials
* Information technology
* Communication services
* Utilities
* Real estate

State Street Global Advisors and Merrill Lynch created and manage 11 exchange-traded funds (ETFs) that are indexed to these 11 sectors of the S&P 500.

This group of 11 ETFs are known as the SPDR (pronounced “spider”) family of funds.

In [ ]:
In this notebook, we will be working with the energy and industrials sector, represented by XLE and XLI. For each of these sectors:
(a) Calculate the expected return in the upcoming year. Use the Capital Asset Pricing Model and the following information:

| Sector | $\beta$ |
|--|--|
| XLE | 1.07 |
| XLI | 1.06 |

* Risk-free rate of return ($𝑅_𝑓$): 2.25%
* Market return ($𝑅_𝑚$): 9%
* Market standard deviation ($\sigma_m$): 15%

## a) Calculating the expected return in the upcoming year. Use the Capital Asset Pricing Model and the following information

### Expected Return

The expected return is the profit or loss an investor anticipates on an investment that has known or anticipated rates
of return (RoR). It is calculated by multiplying potential outcomes by the chances of them occurring and then totaling
these results.

The expected return is a tool used to determine whether an investment has a positive or negative average net outcome. 
The sum is calculated as the expected value (EV) of an investment given its potential returns in different scenarios, 
as illustrated by the following formula:
    
$$Expected Return = SUM(Return_i * Probability_i)$$

where:$i$ indicates each known return and its respective probability in the series

The expected return is the amount of profit or loss an investor can anticipate receiving on an investment.

An expected return is calculated by multiplying potential outcomes by the odds of them occurring and then totaling
these results.

Essentially a long-term weighted average of historical results, expected returns are not guaranteed.

### Limitations of expected return

It is quite dangerous to make investment decisions based on expected returns alone. Before making any buying decisions,
investors should always review the risk characteristics of investment opportunities to determine if the investments align
with their portfolio goals.

### Capital Asset Pricing Model

The Capital Asset Pricing Model (CAPM) describes the relationship between systematic risk and expected return for assets,
particularly stocks. CAPM is widely used throughout finance for pricing risky securities and generating expected returns 
for assets given the risk of those assets and cost of capital.

The formula for calculating the expected return of an asset given its risk is as follows:

$$ER_f = R_f + β_i(E(R_m) - R_f)$$

Investors expect to be compensated for risk and the time value of money. The risk-free rate in the CAPM formula accounts 
for the time value of money. The other components of the CAPM formula account for the investor taking on additional risk.

The beta of a potential investment is a measure of how much risk the investment will add to a portfolio that looks like the
market. If a stock is riskier than the market, it will have a beta greater than one. If a stock has a beta of less than one,
the formula assumes it will reduce the risk of a portfolio.

A stock’s beta is then multiplied by the market risk premium, which is the return expected from the market above the 
risk-free rate. The risk-free rate is then added to the product of the stock’s beta and the market risk premium. 
The result should give an investor the required return or discount rate they can use to find the value of an asset.

The goal of the CAPM formula is to evaluate whether a stock is fairly valued when its risk and the time value of money are 
compared to its expected return.

The CAPM uses the principles of Modern Portfolio Theory to determine if a security is fairly valued. It relies on 
assumptions about investor behaviors, risk and return distributions, and market fundamentals that don’t match reality. 

However, the underlying concepts of CAPM and the associated efficient frontier can help investors understand the 
relationship between expected risk and reward as they make better decisions about adding securities to a portfolio.

In [20]:
# Translating the inputs in the problem statements to code
beta_XLE = 1.07
beta_XLI = 1.06
rf = 2.25
Rm = 9 
sigma_market = 15

## CAPM model

According to the CAPM model, the expected return of the portfolio follows the equation:
$$ E(R_p) = r_f + \beta(E(R_m)-r_f) $$
where,<br>
$R_p$: Portfolio Return<br>
$r_f$: Risk-free rate of Return<br>
$R_m$: Market Rate of Return<br>
$\beta$: Portfolio beta, or exposure<br>
$E(.)$: Expected value function<br>

In [21]:
def expected_return_capm(rf, beta, Rm):
    return rf+beta*(Rm-rf)

In [22]:
expected_return_XLE = expected_return_capm(rf, beta_XLE, Rm)
expected_return_XLI = expected_return_capm(rf, beta_XLI, Rm)

print("Expected Return for XLE :",round(expected_return_XLE,2), "%")
print("Expected Return for XLI :",round(expected_return_XLI,2), "%")

Expected Return for XLE : 9.47 %
Expected Return for XLI : 9.41 %


## b) Calculate the daily and annualized standard deviation of returns for the prior year. Use the MS Excel spreadsheet provided titled “GWP_PTAP_Data.xlsx”. When annualizing daily standard deviation, assume there are 252 trading days in a year.

### Standard Deviation
The standard deviation is a statistic that measures the dispersion of a dataset relative to its mean and is calculated 
as the square root of the variance. It is calculated as the square root of variance by determining the variation between 
each data point relative to the mean. If the data points are further from the mean, there is a higher deviation within the
data set; thus, the more spread out the data, the higher the standard deviation.

A volatile stock has a high standard deviation, while the deviation of a stable blue-chip stock is usually rather low.

As a downside, it calculates all uncertainty as risk, even when it’s in the investor's favor—such as above average returns.

Standard deviation is an especially useful tool in investing and trading strategies as it helps measure market and security
volatility—and predict performance trends. As it relates to investing, for example, one can expect an index fund to have a 
low standard deviation versus its benchmark index, as the fund's goal is to replicate the index.

On the other hand, one can expect aggressive growth funds to have a high standard deviation from relative stock indices, as
their portfolio managers make aggressive bets to generate higher-than-average returns.

#### Properties of Standard Deviations
1) The standard deviation can never be a negative number, due to the way it’s calculated and the fact that it measures a
distance (distances are never negative numbers).

2)The smallest possible value for the standard deviation is 0, and that happens only in contrived situations where every
single number in the data set is exactly the same (no deviation).

3) The standard deviation is affected by outliers (extremely low or extremely high numbers in the data set). That’s because
the standard deviation is based on the distance from the mean. And remember, the mean is also affected by outliers.

4) The standard deviation has the same units as the original data

In [23]:
# Importing the required library for fetching and manipulating the data in code
import pandas as pd
import numpy as np

In [24]:
XLE_data= pd.read_excel ('./GWP_PTAP_Data.xlsx', sheet_name='XLE', header=1)
XLI_data= pd.read_excel ('./GWP_PTAP_Data.xlsx', sheet_name='XLI',header=1)
df = pd.merge(XLE_data, XLI_data, on="Date")
df.columns = ["Date", "XLE", "XLI"]
df = df.set_index("Date")
df.head()

XLE        XLI
Date                            
2017-11-27  67.190002  71.529999
2017-11-28  67.709999  72.620003
2017-11-29  68.080002  73.260002
2017-11-30  69.099998  74.510002
2017-12-01  69.680000  73.589996

In [25]:
daily_returns = df.pct_change()
std_daily_returns = daily_returns.std()

print("Standard Deviation of the Daily Returns of XLE:", round(std_daily_returns["XLE"],4))
print("Standard Deviation of the Daily Returns of XLI:", round(std_daily_returns["XLI"],4))

Standard Deviation of the Daily Returns of XLE: 0.0127
Standard Deviation of the Daily Returns of XLI: 0.0107


In [26]:
# Annualized standard deviation of returns for the prior year.
sigma_annualized = std_daily_returns*np.sqrt(252)
print("Annualized standard deviation (prior year) of XLE:", round(sigma_annualized["XLE"],4))
print("Annualized standard deviation (prior year) of XLI:", round(sigma_annualized["XLI"],4))

Annualized standard deviation (prior year) of XLE: 0.2023
Annualized standard deviation (prior year) of XLI: 0.1703


## c) Calculate the correlation (to two (2) decimal places) between each sector’s return.

Correlation, in the finance and investment industries, is a statistic that measures the degree to which two securities 
move in relation to each other. Correlations are used in advanced portfolio management, computed as the correlation 
coefficient, which has a value that must fall between -1.0 and +1.0.

In finance, the correlation can measure the movement of a stock with that of a benchmark index, such as the Beta.

Correlation measures association, but does not tell you if x causes y or vice versa, or if the association is caused by 
some third (perhaps unseen) factor.

In [27]:
correlation_XLE_XLI = daily_returns.corr()
print("Correlation among the XLE and XLI sector returns is:", round(correlation_XLE_XLI["XLE"]["XLI"],2))

Correlation among the XLE and XLI sector returns is: 0.66


We can see that XLE and XLI are positively correlated and that the correlation is 0.66. This suggests a high level 
of correlation, e.g. a value above 0.5 and close to 1.0.

## Bibliography:

Kenton, Will. "Expected Returns", Investopedia WebPage, 2019. Retrieved from: https://www.investopedia.com/terms/e/expectedreturn.asp

Kenton, Will. "Capital Asset Pricing Model", Investopedia WebPage, 2019. Retrieved from: https://www.investopedia.com/terms/c/capm.asp
        
Kenton, Will. "Standard Deviation", Investopedia WebPage, 2019. Retrieved from: https://www.investopedia.com/terms/s/standarddeviation.asp
        
http://www.yourarticlelibrary.com/statistics-2/correlation-meaning-types-and-its-computation-statistics/92001